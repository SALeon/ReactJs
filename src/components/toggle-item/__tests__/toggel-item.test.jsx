import React from 'react';
import { shallow } from 'enzyme';
import ToggleItem from '../index';

const id = 'id';
const innerComponent = <div>mock</div>;
const anotherId = 'selected';
const toggleListener = jest.fn();

describe('ToggleItem', () => {
  it('state not selected', () => {
    const component = shallow(
      <ToggleItem
        id={id}
        component={innerComponent}
        toggleListener={toggleListener}
        selectedId={anotherId}
      />,
    );
    expect(component.state().isSelected).toBeFalsy();
  });

  it('state selected', () => {
    const component = shallow(
      <ToggleItem
        id={id}
        component={innerComponent}
        toggleListener={toggleListener}
        selectedId={anotherId}
      />,
    );
    expect(component.state().isSelected).toBeFalsy();
  });

  it('snapshot initial component', () => {
    const component = shallow(
      <ToggleItem
        id={id}
        component={innerComponent}
        toggleListener={toggleListener}
        selectedId={id}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('toggle to unselected state', () => {
    const component = shallow(
      <ToggleItem
        id={id}
        component={innerComponent}
        toggleListener={toggleListener}
        selectedId={id}
      />,
    );
    component.setProps({ selectedId: anotherId });
    expect(component.state().isSelected).toBeFalsy();
  });

  it('call click handler of the component ', () => {
    const clickFunc = jest.fn();

    const component = shallow(
      <ToggleItem
        id={id}
        component={innerComponent}
        toggleListener={clickFunc}
        selectedId={anotherId}
      />,
    );
    component.find('.toggle-item').simulate('click');
    expect(clickFunc).toHaveBeenCalledWith(id);
  });
});
