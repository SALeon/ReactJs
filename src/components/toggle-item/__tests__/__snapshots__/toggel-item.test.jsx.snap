// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleItem snapshot initial component 1`] = `
<div
  aria-checked="false"
  className="toggle-item undefined--selected undefined"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="checkbox"
  tabIndex={-1}
>
  <div>
    mock
  </div>
</div>
`;
